spec/input/sources_spec.rb

```diff
@@ -89,4 +91,137 @@
+    let(:output_dir) { "#{DATA_DIR}/output/" }
+
+    it 'should compress the image from input source' do
+      receipt_input = Mindee::Input::Source::PathInputSource.new(input_receipt_path)
+      receipt_input.compress!(quality: 80) # NOTE: base jpg quality is ~81
+
+      FileUtils.mkdir_p(File.dirname("#{output_dir}compress_indirect.jpg"))
+      File.write("#{output_dir}compress_indirect.jpg", receipt_input.io_stream.read)
+
+      initial_file_size = File.size(input_receipt_path)
+      compressed_file_size = File.size(output_dir)
+
+      expect(compressed_file_size).to be < initial_file_size
+    end
+
+    it 'should compress the image with various quality levels' do
+      receipt_input = Mindee::Input::Source::PathInputSource.new(input_receipt_path)
+
+      compresses = [
+        Mindee::Image::ImageCompressor.compress_image(receipt_input.io_stream, quality: 100),
+        Mindee::Image::ImageCompressor.compress_image(receipt_input.io_stream), # default quality
+        Mindee::Image::ImageCompressor.compress_image(receipt_input.io_stream, quality: 50),
+        Mindee::Image::ImageCompressor.compress_image(receipt_input.io_stream, quality: 10),
+        Mindee::Image::ImageCompressor.compress_image(receipt_input.io_stream, quality: 1),
+      ]
+
+      output_files = [
+        "#{output_dir}/compress100.jpg",
+        "#{output_dir}/compress75.jpg",
+        "#{output_dir}/compress50.jpg",
+        "#{output_dir}/compress10.jpg",
+        "#{output_dir}/compress1.jpg",
+      ]
+
+      compresses.zip(output_files).each do |compressed, output_file|
+        File.write(output_file, compressed.read)
+      end
+
+      initial_file_size = File.size(input_receipt_path)
+      rendered_file_sizes = output_files.map { |file| File.size(file) }
+
+      expect(initial_file_size).to be < rendered_file_sizes[0]
+      expect(initial_file_size).to be < rendered_file_sizes[1]
+      expect(rendered_file_sizes[1]).to be > rendered_file_sizes[2]
+      expect(rendered_file_sizes[2]).to be > rendered_file_sizes[3]
+      expect(rendered_file_sizes[3]).to be > rendered_file_sizes[4]
+    end
+
+    after(:each) do
+      FileUtils.rm_f("#{output_dir}/compress100.jpg")
+      FileUtils.rm_f("#{output_dir}/compress75.jpg")
+      FileUtils.rm_f("#{output_dir}/compress50.jpg")
+      FileUtils.rm_f("#{output_dir}/compress10.jpg")
+      FileUtils.rm_f("#{output_dir}/compress1.jpg")
+    end
+  end
+
+  describe 'The PDF text detection method' do
+    it 'should detect text pdf in a PDF file.' do
+      text_input = Mindee::Input::Source::PathInputSource.new("#{DATA_DIR}/file_types/pdf/multipage.pdf")
+      expect(Mindee::PDF::PDFTools.source_text?(text_input.io_stream)).to be(true)
+    end
+
+    it 'should not detect text pdf in an empty PDF file.' do
+      no_text_input = Mindee::Input::Source::PathInputSource.new(
+        "#{DATA_DIR}/products/invoice_splitter/default_sample.pdf"
```

Comment:
this is not an empty file. why not use the blank sample?

Reply:
https://tenor.com/view/the-simpsons-i-dont-know-homer-simpson-bank-gif-2374005963278537110